worker/src/Channel/ChannelSocket.cpp

```diff
@@ -174,124 +160,48 @@ namespace Channel
 	{
 		MS_TRACE_STD();
 
+		size_t msgStart{ 0 };
+
 		// Be ready to parse more than a single message in a single chunk.
 		while (true)
 		{
 			if (IsClosed())
 				return;
 
-			size_t readLen = this->bufferDataLen - this->msgStart;
-			char* msgStart = nullptr;
-			size_t msgLen;
-			int nsRet = netstring_read(
-			  reinterpret_cast<char*>(this->buffer + this->msgStart), readLen, &msgStart, &msgLen);
+			size_t readLen = this->bufferDataLen - msgStart;
 
-			if (nsRet != 0)
+			if (readLen < sizeof(uint32_t))
 			{
-				switch (nsRet)
-				{
-					case NETSTRING_ERROR_TOO_SHORT:
-					{
-						// Check if the buffer is full.
-						if (this->bufferDataLen == this->bufferSize)
-						{
-							// First case: the incomplete message does not begin at position 0 of
-							// the buffer, so move the incomplete message to the position 0.
-							if (this->msgStart != 0)
-							{
-								std::memmove(this->buffer, this->buffer + this->msgStart, readLen);
-								this->msgStart      = 0;
-								this->bufferDataLen = readLen;
-							}
-							// Second case: the incomplete message begins at position 0 of the buffer.
-							// The message is too big, so discard it.
-							else
-							{
-								MS_ERROR_STD(
-								  "no more space in the buffer for the unfinished message being parsed, "
-								  "discarding it");
-
-								this->msgStart      = 0;
-								this->bufferDataLen = 0;
-							}
-						}
-
-						// Otherwise the buffer is not full, just wait.
-						return;
-					}
-
-					case NETSTRING_ERROR_TOO_LONG:
-					{
-						MS_ERROR_STD("NETSTRING_ERROR_TOO_LONG");
-
-						break;
-					}
-
-					case NETSTRING_ERROR_NO_COLON:
-					{
-						MS_ERROR_STD("NETSTRING_ERROR_NO_COLON");
-
-						break;
-					}
-
-					case NETSTRING_ERROR_NO_COMMA:
-					{
-						MS_ERROR_STD("NETSTRING_ERROR_NO_COMMA");
-
-						break;
-					}
-
-					case NETSTRING_ERROR_LEADING_ZERO:
-					{
-						MS_ERROR_STD("NETSTRING_ERROR_LEADING_ZERO");
-
-						break;
-					}
-
-					case NETSTRING_ERROR_NO_LENGTH:
-					{
-						MS_ERROR_STD("NETSTRING_ERROR_NO_LENGTH");
-
-						break;
-					}
-				}
-
-				// Error, so reset and exit the parsing loop.
-				this->msgStart      = 0;
-				this->bufferDataLen = 0;
-
-				return;
+				// Incomplete data.
+				break;
```

Comment:
Why break instead of return? I'm afraid there is super similar logic in `TcpConnection.cpp` and here we are diverging.